brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java

```diff
@@ -0,0 +1,122 @@
+package com.github.kristofa.brave.grpc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.IdConversion;
+import com.github.kristofa.brave.KeyValueAnnotation;
+import com.github.kristofa.brave.ServerRequestAdapter;
+import com.github.kristofa.brave.ServerRequestInterceptor;
+import com.github.kristofa.brave.ServerResponseAdapter;
+import com.github.kristofa.brave.ServerResponseInterceptor;
+import com.github.kristofa.brave.SpanId;
+import com.github.kristofa.brave.TraceData;
+
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public final class BraveGrpcServerInterceptor implements ServerInterceptor {
+
+    private final ServerRequestInterceptor serverRequestInterceptor;
+    private final ServerResponseInterceptor serverResponseInterceptor;
+
+    public BraveGrpcServerInterceptor(Brave brave) {
+        this.serverRequestInterceptor = checkNotNull(brave.serverRequestInterceptor());
+        this.serverResponseInterceptor = checkNotNull(brave.serverResponseInterceptor());
+    }
+
+    @Override
+    public <ReqT, RespT> Listener<ReqT> interceptCall(final MethodDescriptor<ReqT, RespT> method,
+                                                      final ServerCall<RespT> call, final Metadata requestHeaders,
+                                                      final ServerCallHandler<ReqT, RespT> next) {
+        return next.startCall(method, new SimpleForwardingServerCall<RespT>(call) {
+            @Override
+            public void request(int numMessages) {
+                serverRequestInterceptor.handle(new GrpcServerRequestAdapter<>(method, requestHeaders));
+                super.request(numMessages);
+            }
+
+            @Override
+            public void close(Status status, Metadata trailers) {
+                serverResponseInterceptor.handle(new GrpcServerResponseAdapter(status));
+                super.close(status, trailers);
+            }
+        }, requestHeaders);
+    }
+
+    private class GrpcServerRequestAdapter<ReqT, RespT> implements ServerRequestAdapter {
```

Comment:
Same here, why not use `HttpServerRequestAdapter` instead of `ServerRequestAdapter`? 
`public TraceData getTraceData()` implementation is a copy of [HttpServerRequestAdapter](https://github.com/openzipkin/brave/blob/master/brave-http/src/main/java/com/github/kristofa/brave/http/HttpServerRequestAdapter.java).